includes/shortcodes.php

```diff
@@ -124,24 +243,51 @@ function tsml_ui()
         ? $tsml_programs[$tsml_program]['type_descriptions']
         : [];
 
-    //apply settings
-    wp_localize_script(
-        'tsml_ui',
-        'tsml_react_config',
-        array_merge(
-            [
-                'columns' => array_keys($tsml_columns),
-                'conference_providers' => $tsml_conference_providers,
-                'distance_unit' => $tsml_distance_units,
-                'feedback_emails' => array_values($tsml_feedback_addresses),
-                'flags' => $tsml_programs[$tsml_program]['flags'],
-                'strings' => [
-                    $tsml_language => array_merge($tsml_columns, compact('types', 'type_descriptions')),
+    //Load Shortcode values passed, overriding default values
+    $parms = shortcode_atts(array('key' => '', 'value' => '', ), $atts);
+    $key = (isset($parms['key'])) ? sanitize_text_field($parms['key']) : '';
+    $value = (isset($parms['value'])) ? sanitize_text_field($parms['value']) : '';
+    
+    if (!empty($atts)) {
+        //apply settings when parameters are passed
+        wp_localize_script(
+            'tsml_ui',
+            'tsml_react_config',
+            array_merge(
+                [
+                    'defaults' => [$key => explode(',', $value)],
+                    'columns' => array_keys($tsml_columns),
+                    'conference_providers' => $tsml_conference_providers,
+                    'distance_unit' => $tsml_distance_units,
+                    'feedback_emails' => array_values($tsml_feedback_addresses),
+                    'flags' => $tsml_programs[$tsml_program]['flags'],
+                    'strings' => [
+                        $tsml_language => array_merge($tsml_columns, compact('types', 'type_descriptions')),
+                    ],
                 ],
-            ],
-            $tsml_ui_config
-        )
-    );
+                $tsml_ui_config
+            )
+        );
+    } else {
```

Comment:
seems like a lot of code duplication if just `defaults` changes between these implementations. could you try without the else? i think if $types were an empty string then `explode(',', $types)` might be ok

Reply:
Absolutely! If testing shows that works I will remove the code duplication.

Reply:
Duplicate wp_localize_script code removed after testing showed it wasn't needed. Only needed to add 'Any Type' as the default value when no value is received in the $atts array.